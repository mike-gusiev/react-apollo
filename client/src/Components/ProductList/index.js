import React from 'react';
import PropTypes from 'prop-types';

const ProductList = ({products}) => {
    return (
        <ul className='product-list'>
            {products.data && products.data.map(product => (
                <div className="card" key={product.id}>
                    <div className="card-header">{product.name}</div>
                    <div className="card-body">
                        <p className="card-text"><b>Category:</b> {product.category}</p>
                        <p className="card-text"><b>Brand:</b> {product.brand.name}</p>
                    </div>
                </div>
            ))}
        </ul>
    );
};

ProductList.propTypes = {
    products: PropTypes.shape({
        data: PropTypes.arrayOf(PropTypes.shape({
            brand: PropTypes.shape({
                name: PropTypes.string,
                __typename: PropTypes.string
            }),
            name: PropTypes.string,
            id: PropTypes.string,
            category: PropTypes.string,
            __typename: PropTypes.string
        })),
        loading: PropTypes.bool,
        error: PropTypes.string
    })
};

export default ProductList;
