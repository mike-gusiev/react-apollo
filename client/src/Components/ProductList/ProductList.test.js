import React from 'react';
import ProductList from './index';

import renderer from 'react-test-renderer';

test('should render component correctly', () => {
    const products = {
        data: [
            {
                brand: {
                    name: 'test-brand-name',
                    __typename: 'test-brand-type'
                },
                name: 'test-name',
                id: 'test-id',
                category: 'test-category',
                __typename: 'test-typename'
            }
        ],
        loading: true,
        error: ''
    }
    const tree = renderer.create(<ProductList products={products}/>)
        .toJSON()
    expect(tree).toMatchSnapshot();
})

