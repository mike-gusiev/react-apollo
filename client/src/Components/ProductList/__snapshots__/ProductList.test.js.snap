// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render component correctly 1`] = `
<ul
  className="product-list"
>
  <div
    className="card"
  >
    <div
      className="card-header"
    >
      test-name
    </div>
    <div
      className="card-body"
    >
      <p
        className="card-text"
      >
        <b>
          Category:
        </b>
         
        test-category
      </p>
      <p
        className="card-text"
      >
        <b>
          Brand:
        </b>
         
        test-brand-name
      </p>
    </div>
  </div>
</ul>
`;
